### PR TITLE
feat: support com_query insert sql request

### PR DESCRIPTION
**Pull request by wybrobin**
ref: https://github.com/cectc/dbpack/issues/<issueID>

### Ⅰ. Describe what this PR did
support com_query insert sql request

### Ⅱ. Does this pull request fix one issue?
https://github.com/CECTC/dbpack/issues/104

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
Added

### Ⅳ. Describe how to verify it
Run insert sql in com_query

### Ⅴ. Special notes for reviews
No

### CODE REVIEW

**Comment by fatelei** [comment]
LGTM

File: pkg/filter/dt/exec/query_insert.go [hunk 1]
```diff
@@ -69,3 +119,63 @@ func (executor *queryInsertExecutor) GetTableName() string {
...
+	if insertColumns != nil && len(insertColumns) > 0 {
+		for i, columnName := range insertColumns {
+			if strings.EqualFold(tableMeta.GetPKName(), columnName) {
+				return i
+			}
+		}
+	} else {
+		allColumns := tableMeta.Columns
+		var idx = 0
+		for _, column := range allColumns {
```
Comment (dk-lockdown) [inline pkg/filter/dt/exec/query_insert.go]:
```suggestion
		for i, column := range allColumns {
```

File: pkg/filter/dt/exec/query_insert.go [hunk 2]
```diff
@@ -69,3 +119,63 @@ func (executor *queryInsertExecutor) GetTableName() string {
...
+			if strings.EqualFold(tableMeta.GetPKName(), column) {
+				return idx
```
Comment (dk-lockdown) [inline pkg/filter/dt/exec/query_insert.go]:
```suggestion
				return i
```

File: pkg/filter/dt/exec/query_insert.go [hunk 3]
```diff
@@ -69,3 +119,63 @@ func (executor *queryInsertExecutor) GetTableName() string {
...
+			}
+			idx = idx + 1
```
Comment (dk-lockdown) [inline pkg/filter/dt/exec/query_insert.go]:
```suggestion
```

File: pkg/misc/sql.go [hunk 4]
```diff
@@ -46,3 +46,16 @@ func PgsqlAppendInParam(size int) string {
 	fmt.Fprintf(&sb, ")")
 	return sb.String()
 }
+
+func MysqlAppendInParamWithValue(values []interface{}) string {
+	var sb strings.Builder
+	fmt.Fprintf(&sb, "(")
+	for i, value := range values {
+		fmt.Fprintf(&sb, "'%v'", value)
```
Comment (dk-lockdown) [inline pkg/misc/sql.go:54]: if pk type is `[]byte`, there will cause an exception

File: pkg/filter/dt/exec/query_insert.go [hunk 5]
```diff
@@ -69,3 +119,63 @@ func (executor *queryInsertExecutor) GetTableName() string {
...
+	}
+
+	return pkValues, nil
+}
+
+func (executor *queryInsertExecutor) getPKIndex(ctx context.Context) int {
+	insertColumns := executor.GetInsertColumns()
+	tableMeta, _ := executor.GetTableMeta(ctx)
+
+	if insertColumns != nil && len(insertColumns) > 0 {
```
Comment (bohehe) [inline pkg/filter/dt/exec/query_insert.go]:
```suggestion
	if len(insertColumns) > 0 {
```

**Review by bohehe — CHANGES_REQUESTED** [review]

File: pkg/filter/dt/exec/query_insert.go [hunk 6]
```diff
@@ -69,3 +119,62 @@ func (executor *queryInsertExecutor) GetTableName() string {
...
+
+	if insertColumns != nil {
+		for i, columnName := range insertColumns {
+			if strings.EqualFold(tableMeta.GetPKName(), columnName) {
+				return i
+			}
+		}
+	} else {
+		allColumns := tableMeta.Columns
+		var idx = 0
```
Comment (dk-lockdown) [inline pkg/filter/dt/exec/query_insert.go]:
```suggestion
```

File: pkg/filter/dt/exec/query_insert.go [hunk 7]
```diff
@@ -52,8 +54,56 @@ func (executor *queryInsertExecutor) BeforeImage(ctx context.Context) (*schema.T
...
+}
+
+func (executor *queryInsertExecutor) buildAfterImageSql(tableMeta schema.TableMeta, pkValues []interface{}) string {
+	var b strings.Builder
+	b.WriteString("SELECT ")
+	var i = 0
+	columnCount := len(tableMeta.Columns)
+	for _, column := range tableMeta.Columns {
+		b.WriteString(misc.CheckAndReplace(column))
+		i = i + 1
```
Comment (dk-lockdown) [inline pkg/filter/dt/exec/query_insert.go]:
```suggestion
```

File: pkg/filter/dt/exec/query_insert.go [hunk 8]
```diff
@@ -52,8 +54,56 @@ func (executor *queryInsertExecutor) BeforeImage(ctx context.Context) (*schema.T
...
+		if i < columnCount {
```
Comment (dk-lockdown) [inline pkg/filter/dt/exec/query_insert.go]:
```suggestion
		if i < columnCount-1 {
```

File: pkg/filter/dt/exec/query_insert.go [hunk 9]
```diff
@@ -52,8 +54,56 @@ func (executor *queryInsertExecutor) BeforeImage(ctx context.Context) (*schema.T
...
+	}
+	return schema.BuildTextRecords(tableMeta, result), nil
+}
+
+func (executor *queryInsertExecutor) buildAfterImageSql(tableMeta schema.TableMeta, pkValues []interface{}) string {
+	var b strings.Builder
+	b.WriteString("SELECT ")
+	var i = 0
+	columnCount := len(tableMeta.Columns)
+	for _, column := range tableMeta.Columns {
```
Comment (dk-lockdown) [inline pkg/filter/dt/exec/query_insert.go]:
```suggestion
	for i, column := range tableMeta.Columns {
```

File: pkg/filter/dt/exec/query_insert.go [hunk 10]
```diff
@@ -52,8 +54,56 @@ func (executor *queryInsertExecutor) BeforeImage(ctx context.Context) (*schema.T
...
+	if err != nil {
+		return nil, err
+	}
+	return schema.BuildTextRecords(tableMeta, result), nil
+}
+
+func (executor *queryInsertExecutor) buildAfterImageSql(tableMeta schema.TableMeta, pkValues []interface{}) string {
+	var b strings.Builder
+	b.WriteString("SELECT ")
+	var i = 0
```
Comment (dk-lockdown) [inline pkg/filter/dt/exec/query_insert.go]:
```suggestion
```

File: pkg/filter/dt/exec/query_insert.go [hunk 11]
```diff
@@ -69,3 +119,62 @@ func (executor *queryInsertExecutor) GetTableName() string {
...
+	}
+
+	return pkValues, nil
+}
+
+func (executor *queryInsertExecutor) getPKIndex(ctx context.Context) int {
+	insertColumns := executor.GetInsertColumns()
+	tableMeta, _ := executor.GetTableMeta(ctx)
+
+	if insertColumns != nil {
```
Comment (dk-lockdown) [inline pkg/filter/dt/exec/query_insert.go]:
```suggestion
	if len(insertColumns) > 0 {
```

**Review by dk-lockdown — APPROVED** [review]

**Pull request merged**